internal/providers/terraform/aws/cloudfront_distribution.go

```diff
@@ -1,527 +1,38 @@
-			AttributeFilters: []*schema.AttributeFilter{
-				{Key: "location", Value: strPtr(apiRegion)},
-				{Key: "requestType", Value: strPtr("CloudFront-Request-HTTP-Proxy")},
-			},
-		},
-	}
-}
-
-func (c cloudfrontRegionUsage) httpsRequests(regData *regionData) *schema.CostComponent {
-	name := "HTTPS requests"
-
-	apiRegion := regData.priceRegion
-	usageKey := regData.usageKey
-	var quantity *decimal.Decimal
-	if _, ok := c.MonthlyHTTPSReq[usageKey]; ok {
-		quantity = decimalPtr(decimal.NewFromInt(c.MonthlyHTTPSReq[usageKey].Int()))
-	}
-
-	return &schema.CostComponent{
-		Name:            name,
-		Unit:            "10k requests",
-		UnitMultiplier:  decimal.NewFromInt(10000),
-		MonthlyQuantity: quantity,
-		ProductFilter: &schema.ProductFilter{
-			VendorName: strPtr("aws"),
-			Service:    strPtr("AmazonCloudFront"),
-			AttributeFilters: []*schema.AttributeFilter{
-				{Key: "location", Value: strPtr(apiRegion)},
-				{Key: "requestType", Value: strPtr("CloudFront-Request-HTTPS-Proxy")},
-			},
-		},
-	}
-}
-
-var regionShieldMapping = map[string]string{
-	"us-gov-west-1":   "us",
-	"us-gov-east-1":   "us",
-	"us-east-1":       "us",
-	"us-east-2":       "us",
-	"us-west-1":       "us",
-	"us-west-2":       "us",
-	"us-west-2-lax-1": "us",
-	"eu-central-1":    "europe",
-	"eu-west-1":       "europe",
-	"eu-west-2":       "europe",
-	"eu-south-1":      "europe",
-	"eu-west-3":       "europe",
-	"eu-north-1":      "europe",
-	"ap-northeast-1":  "japan",
-	"ap-northeast-2":  "south_korea",
-	"ap-southeast-1":  "singapore",
-	"ap-southeast-2":  "australia",
-	"ap-south-1":      "india",
-	"sa-east-1":       "south_america",
-}
-
-func shieldRequests(d *schema.ResourceData, u *schema.UsageData) *schema.CostComponent {
-	if !d.Get("origin.0.origin_shield.0.enabled").Bool() {
-		return nil
-	}
-
+func newCloudfrontDistribution(d *schema.ResourceData, u *schema.UsageData) *schema.Resource {
 	region := d.Get("region").String()
-	if !d.IsEmpty("origin.0.origin_shield.0.origin_shield_region") {
-		region = d.Get("origin.0.origin_shield.0.origin_shield_region").String()
-	}
-
-	var apiRegion string
-	if v, ok := aws.RegionMapping[region]; ok {
-		apiRegion = v
-	}
-
-	if apiRegion == "" {
-		log.Warnf("Skipping Origin shield HTTP requests for resource %s. Could not find mapping for region %s", d.Address, region)
-		return nil
-	}
-
-	// we need to find the legacy usage key that's defined in the usage file
-	// this is done so we support backwards compatibility and don't nuke old usage files.
-	var usageKey string
-	if v, ok := regionShieldMapping[region]; ok {
-		usageKey = v
-	}
+	isOriginShieldEnabled := d.Get("origin.0.origin_shield.0.enabled").Bool() || false
+	isSSLSupportMethodVIP := d.Get("viewer_certificate.0.ssl_support_method").String() == "vip"
+	hasLoggingConfigBucket := !d.IsEmpty("logging_config.0.bucket")
+	hasFieldLevelEncryptionID := !d.IsEmpty("default_cache_behavior.0.field_level_encryption_id")
```

Comment:
```suggestion
	hasFieldLevelEncryptionID := d.Get("default_cache_behavior.0.field_level_encryption_id").String() != ""
```
Should we check that this is not an empty string as well?

Reply:
`IsEmpty` checks that as well 👍